Remove dependency on contextlib import AsyncExitStack

jmcgrath207 opened the pull request:
Removed dependency on contextlib import AsyncExitStack 

Tested with `tests.py` with environment var `TESTING_USE_AWS_KINESIS=0`

jmcgrath207 commented:
fixed missing await, ran test with tests.py with environment var `TESTING_USE_AWS_KINESIS=1`

30 out of 31 test pass, `test_consumer_consume_fetch_limit` fails

hampsterx commented:
hi @jmcgrath207 looks good, just scratching head wondering why I used AsyncExitStack in the first place :P

hampsterx commented:
> test_consumer_consume_fetch_limit

```
@skipUnless(
        TESTING_USE_AWS_KINESIS, "Requires TESTING_USE_AWS_KINESIS flag to be set"
    )
    async def test_consumer_consume_fetch_limit(self):
```

ahh that test runs on aws, those tests are a bit fragile as throttling does not always occur very precisely.

jmcgrath207 commented:
haha don't feel bad I was doing that for the longest time, until I found this in the dash repo. So I am a no good code stealer :sunglasses: 

https://github.com/dask/s3fs/blob/86e8dbda126d9bebfe4dbc7311b6851d1ec7a90c/s3fs/core.py#L316


Anyways I found the issue with that test, well at least for me. I made the `test_consumer_consume_fetch_limit` test more aggressive so it would trigger those 

I also forgot to add a test in my last pull request for the json list aggregator since I only added the processor test, so I added this `test_list_aggregator` in test.py

Lastly, I removed the creation of the async loop since it's not needed anymore for python 3.7 and 3.8 and it's going to be deprecated in upcoming python versions.

I retested all of this with tox for python 3.7 and 3.8 and everything is working except for the black test.

Next time I will do a better job keeping my pull requests in scope :sweat_smile:

hampsterx commented on test-requirements.txt:
```diff
@@ -5,3 +5,5 @@ nose==1.3.7
 pinocchio==0.4.2
 python-dotenv==0.9.1
 asynctest==0.12.3
+pluggy==0.13.1
```
err what was this added for?

jmcgrath207 replied:
@hampsterx So I had to add that for Tox to make it work in pycharm. If we need to pop that out, that's fine.
